docs/development/new-cloud-provider-botanist.md

```diff
@@ -0,0 +1,61 @@
+# Botanist Implementation for New Cloud Provider
+
+This document describes how to extend Botanist to support a new cloud provider. There is a significant number of steps necessary.
+
+## Steps
+
+The implementation consists of several parts:
+
+1. Update the helm chart configurations
+2. Update the Gardener golang code to support the cloud provider.
+
+### Helm charts
+
+1. If your implementation is not in-tree to Kubernetes as a cloud provider, add the cloud-controller-manager (CCM) and cloud-storage-interface (CSI) images to [charts/images.yaml](../../charts/images.yaml). This is _just_ a manifest of all _potential_ images to be used in Gardener. The images are referenced in Gardener by the unique `name` key in the file, and can be used elsewhere in Gardener.
+    * Add the provider-specific CCM with a unique name across Gardener
+    * Add the provider-specific CSI with a unique name across Gardener
+2. Update config files for the seed control plane charts in [charts/seed-controlplane/charts/](../../charts/seed-controlplane/charts/):
+    1. If necessary, update the seed control plane chart for the cloud controller manager in `machine-controller-manager/templates/cloud-controller-manager.yaml`
+    2. If necessary, update the default values for the cloud controller manager in `machine-controller-manager/values.yaml`
+    3. Ensure the cloud provider is listed as an option for the cloud volume plugin in the relevant `if contains` statement in `kube-controller-manager/templates/_helpers.tpl`
+3. Create config files for the seed machines charts in [charts/seed-machines/charts](../../charts/seed-machines/charts)::
```

Comment:
double `:` at end of line

Reply:
Mention that the <provider>MachineClass CRD must be deployed in `charts/seed-bootstrap`

Reply:
> double : at end of line

Fixed.

> Mention that the MachineClass CRD must be deployed in charts/seed-bootstrap

I don't understand. Where is that in #626 ?

Reply:
It's not there because Alicloud support is part of MCM very long. We updated MCM versions previously and needed to install the CRD already back then (although it was not used, of course).

Reply:
Fair enough. But I still don't understand what needs to be done here. "MachineClass CRD must be deployed in charts/seed-bootstrap" doesn't mean anything to me, and I don't have an example to point to.

Reply:
1. https://github.com/gardener/gardener/commit/c9f5b4f57be0b9971006a085695be09c9e270c54#diff-7675e9272d765f5ba94432791fa80f21
1. https://github.com/gardener/gardener/commit/c9f5b4f57be0b9971006a085695be09c9e270c54#diff-a7196cdcf6f87e6b16886dfcd372bd7d

Reply:
I can work with that.

Reply:
Done.